platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -864,9 +864,19 @@ public TransformUpstreamDependenciesResolverFactory getDependenciesResolverFacto
     }
 
     @Override
-    public void resetResolutionState() {
+    public <T> T withResetResolutionState(Factory<T> factory) {
         warnOnInvalidInternalAPIUsage("resetResolutionState()", ProperMethodUsage.RESOLVABLE);
```

Comment:
The string here should be changed to the new name as well.

Reply:
Good catch.  Fixed.